src/fsharp/fsi/fsi.fs

```diff
@@ -2755,7 +2755,13 @@ type FsiEvaluationSession (fsi: FsiEvaluationSessionHostConfig, argv:string[], i
     let currentDirectory = Directory.GetCurrentDirectory()
     let tryGetMetadataSnapshot = (fun _ -> None)
 
-    let defaultFSharpBinariesDir = FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(FSharpEnvironment.tryCurrentDomain()).Value
+    let defaultFSharpBinariesDir = 
+        try
```

Comment:
If the problem is that `FSharpEnvironment.tryCurrentDomain()` returns a `null` then please write something like this:

```fsharp
    let defaultFSharpBinariesDir = 
        let fallback() =
            let d = Assembly.GetExecutingAssembly()
            Path.GetDirectoryName d.Location
        match FSharpEnvironment.tryCurrentDomain() with
        | null -> fallback()
        | dom -> 
            match FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(dom) with
            | None -> fallback()
            | Some dir -> dir

```

Reply:
Done.